iree/compiler/Codegen/LLVMGPU/test/distribute_to_thread.mlir

```diff
@@ -85,37 +88,37 @@ hal.executable.variant @cuda, target = #hal.executable.target<"cuda", "cuda-nvpt
 
 //   CHECK-LABEL: hal.executable @dot_dispatch_0
 //         CHECK:   hal.executable.variant @cuda
-//         CHECK:  memref.global "private" @{{.*}} : memref<4x256xf32, 3>
-//         CHECK:  memref.global "private" @{{.*}} : memref<2x4xf32, 3>
-//     CHECK-DAG:  %[[C1024:.+]] = constant 1024 : index
+//         CHECK:  memref.global "private" @{{.*}} : memref<4x128xf32, 3>
+//         CHECK:  memref.global "private" @{{.*}} : memref<16x4xf32, 3>
 //     CHECK-DAG:  %[[C0:.+]] = constant 0 : index
 //     CHECK-DAG:  %[[C1:.+]] = constant 1 : index
-//     CHECK-DAG:  %[[C2:.+]] = constant 2 : index
 //     CHECK-DAG:  %[[C4:.+]] = constant 4 : index
-//     CHECK-DAG:  %[[C256:.+]] = constant 256 : index
+//     CHECK-DAG:  %[[C16:.+]] = constant 16 : index
+//     CHECK-DAG:  %[[C128:.+]] = constant 128 : index
+//     CHECK-DAG:  %[[C1024:.+]] = constant 1024 : index
 //         CHECK:  scf.for %[[K:.+]] = %[[C0]] to %[[C1024]] step %[[C4]] {
-//         CHECK:  gpu.barrier
-//         CHECK:  %[[TIDX:.+]] = "gpu.thread_id"() {dimension = "x"} : () -> index
-//         CHECK:  scf.for %[[IND:.+]] = %[[TIDX]] to %[[C2]] step %[[C2]] {
-//     CHECK-DAG:    %[[SRC:.+]] = memref.subview %{{.*}}[%[[IND]], 0] [1, 4] [1, 1] : memref<2x4xf32, #{{.*}}> to memref<1x4xf32, #{{.*}}>
-//     CHECK-DAG:    %[[DST:.+]] = memref.subview %{{.*}}[%[[IND]], 0] [1, 4] [1, 1] : memref<2x4xf32, #{{.*}}, 3> to memref<1x4xf32, #{{.*}}, 3>
-//         CHECK:    linalg.copy(%[[SRC]], %[[DST]]) {__internal_linalg_transform__ = "vectorize"} : memref<1x4xf32, #{{.*}}>, memref<1x4xf32, #{{.*}}, 3>
-//         CHECK:    }
-//         CHECK:  %[[TIDX:.+]] = "gpu.thread_id"() {dimension = "x"} : () -> index
-//         CHECK:  scf.for %[[IND0:.+]] = %[[C0]] to %[[C4]] step %[[C1]] {
-//         CHECK:    scf.for %[[IND1:.+]] = %{{.*}} to %[[C256]] step %[[C256]] {
-//     CHECK-DAG:      %[[SRC:.+]] = memref.subview %{{.*}}[%[[IND0]], %[[IND1]]] [1, 4] [1, 1] : memref<4x256xf32, #{{.*}}> to memref<1x4xf32, #{{.*}}>
-//     CHECK-DAG:      %[[DST:.+]] = memref.subview %{{.*}}[%[[IND0]], %[[IND1]]] [1, 4] [1, 1] : memref<4x256xf32, #{{.*}}, 3> to memref<1x4xf32, #{{.*}}, 3>
+//         CHECK:    gpu.barrier
+//         CHECK:    %[[TIDX:.+]] = "gpu.thread_id"() {dimension = "x"} : () -> index
+//         CHECK:    scf.for %[[IND:.+]] = %[[TIDX]] to %[[C16]] step %[[C16]] {
+//     CHECK-DAG:      %[[SRC:.+]] = memref.subview %{{.*}}[%[[IND]], 0] [1, 4] [1, 1] : memref<16x4xf32, #{{.*}}> to memref<1x4xf32, #{{.*}}>
+//     CHECK-DAG:      %[[DST:.+]] = memref.subview %{{.*}}[%[[IND]], 0] [1, 4] [1, 1] : memref<16x4xf32, #{{.*}}, 3> to memref<1x4xf32, #{{.*}}, 3>
```

Comment:
Sure make sense. This particular test was giving me lot of headache. Ill just set the config, etc. explicitly and make that work. It depends on what your intent on these tests were. The earlier pass was not setting the config and using the default? I wasnt sure what was happening with this tests because `LLVMGPUTileAndDistribute` was setting the config and also doing tile and distribute. Ill take a run at this test again.

Reply:
Yes the test is meant to test the transformation and not the config. We need a simpler way to test the config otherwise changing it is a nightmare as we need to completely rewrite the test :( If something is using the default feel free to add the config.

Reply:
I updated the PR to add `gpu_set_num_workgroups.mlir` that tests the config. Also changed the `distribute_to_threads.mlir` to test to have preset configurations.
(Have to say that was not as easy as I thought it would be)